ssl/ssl_lib.c

```diff
@@ -4404,7 +4404,7 @@ EVP_MD_CTX *ssl_replace_hash(EVP_MD_CTX **hash, const EVP_MD *md)
 {
     ssl_clear_hash_ctx(hash);
     *hash = EVP_MD_CTX_new();
-    if (*hash == NULL || (md && EVP_DigestInit_ex(*hash, md, NULL) <= 0)) {
+    if (*hash == NULL || (md && !EVP_DigestInit_ex(*hash, md, NULL))) {
```

Comment:
Hmmm... Do we call this function with a NULL md, ever? Is it reasonable to expect it to be NULL? If not, then I don't see why we're testing it...

Reply:
Hmmm, good question... Its seems to be the only place in the entire source code where this is checked. 
```
~/src/openssl$ rgrep EVP_DigestInit | grep -F '&&'
ssl/ssl_lib.c:4407:    if (*hash == NULL || (md && !EVP_DigestInit_ex(*hash, md, NULL))) {
```
Probably you're right. I will remove the check and see what the tests say about it.

Reply:
Just noticed right now that the [doc comment]( https://github.com/mspncp/openssl/blob/226e65c417e234d5cdc6a5b19b7c5a2197d11920/ssl/ssl_lib.c#L4401) says that the md is optional. So even if nobody would use this feature we still couldn't change it in this release.